hazelcast/src/main/java/com/hazelcast/instance/impl/JetExtension.java

```diff
@@ -95,24 +108,22 @@ public void beforeClusterStateChange(ClusterState requestedState) {
         }
     }
 
-    public void onClusterStateChange(ClusterState ignored) {
-        if (activated) {
+    public void onClusterStateChange() {
+        if (tryActivate()) {
             jetServiceBackend.getJobCoordinationService().clusterChangeDone();
         }
     }
 
-    public void onClusterVersionChange(Version newVersion) {
-        if (!activated && isAfterStartCalled && newVersion.isGreaterOrEqual(Versions.V5_0)) {
-            // Activate Jet after rolling upgrade in which the cluster
-            // version is upgraded from 4.x to 5.0
-            activated = true;
-            jetServiceBackend.getJobCoordinationService().startScanningForJobs();
+    public void onClusterVersionChange() {
+        // Activate Jet after rolling upgrade in which the cluster
+        // version is upgraded from 4.x to 5.0
+        if (!activated.get() && tryActivate()) {
             logger.info("Jet is enabled after the cluster version upgrade.");
```

Comment:
But, this log can be missed when jet is activated after the cluster version upgrade and this activation is performed on another place such as on getJet.